test/chainrules.jl

```diff
@@ -13,6 +13,27 @@
         end
     end
 
+    @testset "xexpx" begin
+        # regular branch
+        test_scalar(xexpx, randn())
+        # special cases (manually since FiniteDifferences/ChainRulesTestUtils fails at -Inf)
+        @test @inferred(frule((NoTangent(), 1), xexpx, -Inf)) === (0.0, 0.0)
```

Comment:
I think it's better to use `randn` to ensure that the result is independent of the tangent:
```suggestion
        @test @inferred(frule((NoTangent(), randn()), xexpx, -Inf)) === (0.0, 0.0)
```

Reply:
That can fail because the tangent can be `-0.0` or `0.0`, depending on the sign output by `randn()`.

Reply:
I can use `rand()` though.